adapters/info_test.go

```diff
@@ -69,57 +71,125 @@ func TestImpFiltering(t *testing.T) {
+				Imp: []openrtb.Imp{
+					{
+						ID:    "imp-1",
+						Video: &openrtb.Video{},
+					},
+					{
+						Native: &openrtb.Native{},
+					},
+					{
+						ID:     "imp-2",
+						Video:  &openrtb.Video{},
+						Native: &openrtb.Native{},
+					},
+					{
+						Banner: &openrtb.Banner{},
+					},
+				},
+				Site: &openrtb.Site{},
+			},
+			expectedErrors: []error{
+				&errortypes.BadInput{Message: "request.imp[1] uses native, but this bidder doesn't support it"},
+				&errortypes.BadInput{Message: "request.imp[2] uses native, but this bidder doesn't support it"},
+				&errortypes.BadInput{Message: "request.imp[3] uses banner, but this bidder doesn't support it"},
+				&errortypes.BadInput{Message: "request.imp[1] has no supported MediaTypes. It will be ignored"},
+				&errortypes.BadInput{Message: "request.imp[3] has no supported MediaTypes. It will be ignored"},
+			},
+			expectedImpLen: 2,
+		},
+		{
+			description: "All imps with correct media type, MakeRequest() call expected",
+			inBidRequest: &openrtb.BidRequest{
+				Imp: []openrtb.Imp{
+					{ID: "imp-1", Video: &openrtb.Video{}},
+					{ID: "imp-2", Video: &openrtb.Video{}},
+				},
+				Site: &openrtb.Site{},
+			},
+			expectedErrors: nil,
+			expectedImpLen: 2,
 		},
-		Site: &openrtb.Site{},
-	}, &adapters.ExtraRequestInfo{})
-	if !assert.Len(t, errs, 6) {
-		return
 	}
-	assert.EqualError(t, errs[0], "request.imp[1] uses native, but this bidder doesn't support it")
-	assert.EqualError(t, errs[1], "request.imp[2] uses native, but this bidder doesn't support it")
-	assert.EqualError(t, errs[2], "request.imp[3] uses banner, but this bidder doesn't support it")
-	assert.EqualError(t, errs[3], "request.imp[1] has no supported MediaTypes. It will be ignored")
-	assert.EqualError(t, errs[4], "request.imp[3] has no supported MediaTypes. It will be ignored")
-	assert.EqualError(t, errs[5], "mock MakeRequests error")
-	assert.IsType(t, &errortypes.BadInput{}, errs[0])
-	assert.IsType(t, &errortypes.BadInput{}, errs[1])
-	assert.IsType(t, &errortypes.BadInput{}, errs[2])
-	assert.IsType(t, &errortypes.BadInput{}, errs[3])
-	assert.IsType(t, &errortypes.BadInput{}, errs[4])
 
-	req := bidder.gotRequest
-	if !assert.Len(t, req.Imp, 2) {
-		return
+	for _, test := range testCases {
+		actualAdapterRequests, actualErrs := constrained.MakeRequests(test.inBidRequest, &adapters.ExtraRequestInfo{})
+
+		// Assert the request.Imp slice was correctly filtered and if MakeRequest() was called by asserting
+		// the corresponding error messages were returned
+		for i, expectedErr := range test.expectedErrors {
+			assert.EqualError(t, expectedErr, actualErrs[i].Error(), "Test failed. Error[%d] in error list mismatch: %s", i, test.description)
+		}
+
+		// Extra MakeRequests() call check: our mockBidder returns an adapter request for every imp
+		assert.Equal(t, test.expectedImpLen, len(actualAdapterRequests), "Test failed. Incorrect lenght of filtered imps: %s", test.description)
```

Comment:
@SyntaxNode let me know if you agree with this "extra" check. Strictly speaking, we don't need it. Do you want me to remove it?

Reply:
I think it's fine to keep. There's a few test cases with expected imps. Consider using `assert.Len` though.